pkgs/applications/misc/furtherance/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform, appstream-glib, desktop-file-utils
+, glib, libadwaita, meson, ninja, pkg-config, python3, wrapGAppsHook4, dbus
+, gtk4, sqlite }:
+
+stdenv.mkDerivation rec {
+  pname = "furtherance";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "lakoliu";
+    repo = "Furtherance";
+    rev = "v${version}";
+    sha256 = "oIzwFPJUY+OwT2B6n7NPg8bMxEd4D0MKPe3GQxrXmEU=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "FEvRcUzV5gk9NU/pMOLFLsMfZq6nMsKLRvpFR14W/eU=";
+  };
+
+  nativeBuildInputs = [
+    appstream-glib
+    desktop-file-utils
+    meson
+    ninja
+    pkg-config
+    python3
+    rustPlatform.cargoSetupHook
+    rustPlatform.rust.cargo
+    rustPlatform.rust.rustc
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [ dbus glib gtk4 libadwaita sqlite ];
+
+  enableParallelBuilding = true;
+
+  preInstall = ''
+    chmod +x ../build-aux/meson/postinstall.py
+    patchShebangs ../build-aux/meson/postinstall.py
+    substituteInPlace ../build-aux/meson/postinstall.py \
+      --replace /usr/local $out
+    substituteInPlace ../build-aux/meson/postinstall.py \
+      --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
```

Comment:
If you are up to it, it would be nice to switch to Meson’s new post install mechanism upstream so this whole thing can be dropped on next release. Example: https://gitlab.gnome.org/GNOME/d-spy/-/merge_requests/6/diffs

Reply:
This is something that would need to be modified on the source code of furtherance right? I'm not the developer, just packaging it for nix!

Reply:
Right. Often, upstream developers are not aware of the constantly evolving build system best practices and this is where people packaging can help. Making changes upstream can lead to shorter and cleaner packages for Nixpkgs and other distributions.

Reply:
I've made a [patch for this](https://gist.github.com/winterqt/a596b439016e65b7cceee3a299392128), and I've confirmed it builds. @CaptainJawZ do you mind testing that this runs as expected? You can test this by adding it to `patches` (using `fetchpatch` or a local path) and deleting that entire `postinstall` patch block. If everything works as expected, I'll submit the patch upstream.

Reply:
Thank you so much for the help! I'll try it out right now, I made a rebase with all the suggested changes except this one because postInstall doesn't let the app to build, I'll test out this patch.

Reply:
Alright, the patch works perfectly thank you so very much for the contribution! this also helps me prevent what would've been a problem with the other package I'm working on, now I know how to properly patch!

with this, I think all the suggestions have been implemented.

I have one question tho, version 1.5.3 is out now, should I wait till this current init is merged into the repo, or adjust it so this builds 1.5.3 instead of 1.5.1? 

It doesn't have any major changes other than a new icon and some translations to french, so in my opinion it can wait.

Reply:
> or adjust it so this builds 1.5.3 instead of 1.5.1?

This would be preferable IMO, considering all you should have to do is change version + hashes + commit message.

Reply:
working on it then!